### PR TITLE
LPD-39261 Template engine is not defining the nonceAttribute when rendering email notifications

### PR DESCRIPTION
**Pull request by izaera**
# :green_circle: This PR is safe for forwarding after code is reviewed

QA analysis -> https://github.com/liferay-frontend/liferay-portal/pull/4489#issuecomment-2418655156

### CODE REVIEW

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:test:sf

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:report:41447323

**Comment by izaera** [comment]
Test failures don't seem related to the PR but I have rebased with today's master and I'm going to run them again.

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:report:43035250

**Comment by izaera** [comment]
QA analysis:

1. **PortalLogAssertorTest#testScanXMLLog: {code=404. msg=The requested resource [/html/sXcC2CHf] is not available. uri=/html/sXcC2CHf}**: does not seem related to this PR
2. **PortalLogAssertorTest#testScanXMLLog: Unable to bind to the LDAP server javax.naming.CommunicationException: localhost:10389**: unrelated to the PR

PR seems safe to me :white_check_mark:

File: modules/apps/portal-template/portal-template-test/src/testIntegration/java/com/liferay/portal/template/test/TemplateContextHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		Map<String, Object> contextObjects = new HashMap<>();
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(
+			"com.liferay.portal.security.content.security.policy.internal." +
+				"ContentSecurityPolicyNonceManager#NONCE",
+			"TEST_NONCE");
```
Comment (dsanz) [inline modules/apps/portal-template/portal-template-test/src/testIntegration/java/com/liferay/portal/template/test/TemplateContextHelperTest.java:63]: Current api to set nonce does not allow us to set a test nonce, so I understand this is the only way we can test the override in `prepare()`. Nevertheless this part will fail if API implementation changes the way to set/store the nonce in the request even if prepare does its job. I'm fine with this, we can make it more robust when needed

**Review by dsanz — COMMENTED** [review]
LGTM ;) pls see comments inline

**Review by bryceosterhaus — COMMENTED** [review]
LGTM, other than Dani's comments

File: modules/apps/portal-template/portal-template-test/src/testIntegration/java/com/liferay/portal/template/test/TemplateContextHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	public void testHelperRestrictedUtilitiesHaveBlankNonceAttribute()
+		throws Exception {
+
+		TemplateContextHelper templateContextHelper =
+			new TemplateContextHelper();
+
+		Map<String, Object> helperUtilities =
+			templateContextHelper.getHelperUtilities(true);
+
+		Assert.assertTrue(helperUtilities.containsKey("nonceAttribute"));
```
[the remark this answers is not in the document]
Comment (izaera) [reply modules/apps/portal-template/portal-template-test/src/testIntegration/java/com/liferay/portal/template/test/TemplateContextHelperTest.java:47]: According to the test's name, yes...

Let me check why I did it this way :thinking:

[the remark this answers is not in the document]
Comment (izaera) [reply modules/apps/portal-template/portal-template-test/src/testIntegration/java/com/liferay/portal/template/test/TemplateContextHelperTest.java]: fixed :ok_hand:

**Comment by izaera** [comment]
ci:forward

**Pull request closed without merging**